File: win-prog/ProgWinCS/Brushes and Pens/LineJoins/LineJoins.cs
//----------------------------------------
// LineJoins.cs © 2001 by Charles Petzold
//----------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class LineJoins: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new LineJoins());
     }
     public LineJoins()
     {
          Text = "Line Joins: Miter, Bevel, Round, MiterClipped";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Pen     penNarrow = new Pen(clr);
          Pen     penWide   = new Pen(Color.Gray, cx / 16);
          Point[] apt       = { new Point(1 * cx / 32, 1 * cy / 8), 
                                new Point(4 * cx / 32, 6 * cy / 8),
                                new Point(7 * cx / 32, 1 * cy / 8) };

          for (int i = 0; i < 4; i++)
          {
               penWide.LineJoin = (LineJoin) i;

               grfx.DrawLines(penWide, apt);
               grfx.DrawLines(penNarrow, apt);
               grfx.TranslateTransform(cx / 4, 0);
          }
     }
}

File: win-prog/ProgWinVB/Brushes and Pens/LineJoins/LineJoins.vb
'------------------------------------------
' LineJoins.vb (c) 2002 by Charles Petzold
'------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class LineJoins
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New LineJoins())
    End Sub

    Sub New()
        Text = "Line Joins: Miter, Bevel, Round, MiterClipped"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim pnNarrow As New Pen(clr)
        Dim pnWide As New Pen(Color.Gray, cx \ 16)
        Dim apt() As Point = {New Point(1 * cx \ 32, 1 * cy \ 8), _
                              New Point(4 * cx \ 32, 6 * cy \ 8), _
                              New Point(7 * cx \ 32, 1 * cy \ 8)}
        Dim i As Integer

        For i = 0 To 3
            pnWide.LineJoin = CType(i, LineJoin)
            grfx.DrawLines(pnWide, apt)
            grfx.DrawLines(pnNarrow, apt)
            grfx.TranslateTransform(cx \ 4, 0)
        Next i
    End Sub
End Class
